providers/google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/domain/Network.java

```diff
@@ -38,24 +46,39 @@
 
    public abstract String name();
 
+   public abstract NetworkType type();
+
    @Nullable public abstract String description();
 
    /**
     * The range of internal addresses that are legal on this network. This range is a CIDR
     * specification, for example: {@code 192.168.0.0/16}.
     */
-   public abstract String rangeIPv4();
+   @Nullable public abstract String rangeIPv4();
 
    /**
     * This must be within the range specified by IPv4Range, and is typically the first usable address in that range.
     * If not specified, the default value is the first usable address in IPv4Range.
     */
    @Nullable public abstract String gatewayIPv4();
 
-   @SerializedNames({ "id", "creationTimestamp", "selfLink", "name", "description", "IPv4Range", "gatewayIPv4" })
+   @Nullable public abstract List<URI> subnetworks();
+
+   public static Network create(String id, Date creationTimestamp, URI selfLink, String name, String description, String rangeIPv4,
+                                String gatewayIPv4) {
+      return new AutoValue_Network(id, creationTimestamp, selfLink, name, NetworkType.LegacyNetwork, description,
+              rangeIPv4, gatewayIPv4, null);
+   }
```

Comment:
I thought this would cause problems if the customer was creating the network in their code using this constructor. That's why I moved the automatic to the more generic but I kept the original one in place.